src/main/scala/com/ing/wbaa/gargoyle/proxy/provider/AuthorizationProviderRanger.scala

```diff
@@ -36,30 +36,55 @@ trait AuthorizationProviderRanger extends LazyLogging {
   def rangerPluginForceInit: RangerBasePlugin = rangerPlugin
 
   /**
-   * Check authorization with Ranger. Currently we deny any requests not to a specific bucket (e.g. listBuckets)
+   *  Check authorization with Ranger. Operations like list-buckets or create, delete bucket must be
+   *  enabled in configuration. They are disabled by default
    */
   def isUserAuthorizedForRequest(request: S3Request, user: User): Boolean = {
-    request.bucket match {
-      case Some(bucket) =>
-        import scala.collection.JavaConverters._
+    request match {
```

Comment:
It might be more clear to put these nested if/match checks on the highest level. So sth like:

request match {
  case S3Request(_, Some(bucket), Some(bucketObject), accessType) if bucketObject.length > 1 =>
     //object operation

  case S3Request(_, None, None, accessType) if accessType == Read =>
    //list objects
}